geoh5py/shared/concatenation/concatenator.py

```diff
@@ -472,30 +473,40 @@ def remove_children(self, children: list | Concatenated):
 
             self.remove_entity(child)
 
-    def remove_entity(self, entity: Concatenated):
+    def remove_entity(self, entity: Concatenated | ConcatenatedPropertyGroup):
         """Remove a concatenated entity."""
 
+        parent = entity.parent
         if isinstance(entity, ConcatenatedData):
             # Remove the rows of data and index
             self.update_array_attribute(entity, entity.name, remove=True)
+            # Remove the data from the group
+
+            if entity.property_group is not None:
+                entity.property_group.remove_properties([entity])
+
             # Remove from the concatenated Attributes
-            parent_attr = self.get_concatenated_attributes(entity.parent.uid)
+            parent_attr = self.get_concatenated_attributes(parent.uid)
             name = entity.name
             del parent_attr[f"Property:{name}"]
-        elif isinstance(entity, ConcatenatedObject):
-
-            for child in entity.children.copy():
-                self.remove_entity(child)
-
-            if entity.property_groups is not None:  # type: ignore
-                self.update_array_attribute(entity, "property_groups", remove=True)
 
+        elif isinstance(entity, ConcatenatedObject):
+            # First remove the children
+            entity.remove_children(entity.children.copy())
```

Comment:
???

This line is strange. Don't we want to drop the children based on uid, not on "copy"?

Reply:
This is just a copy of the list. Issues arise otherwise in the next methods where data are removed from the same children list (list modified within the loop)

Reply:
ok with that.
Marking it as resolved

Reply:
ok with that.
Marking it as resolved